src/action/network.rs

```diff
@@ -0,0 +1,297 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! A handler and associated types for the network connection action.
+//!
+//! The network connection action lists all the opened connections on the
+//! client and the information about them (e.g. type of the connection, source
+//! and target IP addresses, the process that owns the connection).
+
+use std::fmt::{Display, Formatter};
+use std::net::IpAddr;
+
+use log::error;
+use sysinfo::{System, SystemExt, Process, ProcessExt};
+use netstat2::{self, ProtocolSocketInfo, TcpState};
+use rrg_proto::{
+    ListNetworkConnectionsArgs,
+    NetworkConnection,
+    NetworkEndpoint,
+    network_connection::{Family, Type, State},
+};
+
+use crate::session::{self, Session};
+
+#[derive(Debug)]
+struct Error(netstat2::error::Error);
```

Comment:
I know this type is not exposed, but I still prefer to put documentation to everything.

Reply:
I didn't document the `Error`, because I looked at the example action and saw that `Error` is not documented there. So, is it OK to leave this type undocumented, or is it a bug that `Error` in `actions/startup.rs` is not documented?

Reply:
Oh, thanks for pointing this out. I would not call it a "bug", but it is definitely an overlook on my part—the documentation should be there.

Reply:
Addressed in d9700b1fc257127cee293d6350b80c1151095342.